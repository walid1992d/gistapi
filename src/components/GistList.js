import React, { useEffect, useState, useContext } from 'react'
import { GistContext } from '../contexts/gistContext';
import Gist from './Gist';

const GistList = () => {
    const {gistListState} = useContext(GistContext);

  
return (
<>
{
    gistListState.map(element => {
        return <Gist key={element.url} gist={element}></Gist>;
    })
}
</>
);

}

export default GistList
