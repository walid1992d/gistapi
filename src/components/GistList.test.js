import { render } from '@testing-library/react';
import React from 'react';
import { GistContext } from '../contexts/gistContext';
import GistList from './GistList';
jest.mock('./Gist', ()=> {
    return ()=> <div className="gist"></div>
});
describe("GistList Component", ()=> {
    const list = [{url: '1'}, {url:'2'}]
    it("should render correctly", () => {
        const {container} = render(
            <GistContext.Provider value={{gistListState: list, setGistList:jest.fn()}}>
                <GistList></GistList>
            </GistContext.Provider>
        )
        expect(container).toMatchSnapshot();
        expect(container.querySelectorAll('.gist').length).toEqual(2)
    });
    
})