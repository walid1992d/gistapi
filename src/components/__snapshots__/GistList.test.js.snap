// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistList Component should render correctly 1`] = `
<div>
  <div
    class="gist"
  />
  <div
    class="gist"
  />
</div>
`;
